AnkiDroid/jacoco.gradle

```
@@ -11,16 +13,59 @@ android {
     }
 }
 
+@Memoized
+Properties getLocalProperties() {
+    final propertiesFile = project.rootProject.file("local.properties")
+
+    final properties = new Properties()
+
+    if (propertiesFile.exists()) {
+        properties.load(propertiesFile.newDataInputStream())
+    }
+
+    return properties
+}
+
+def openReport(htmlOutDir) {
+    final reportPath = "$htmlOutDir/index.html"
+
+    println "HTML Report: $reportPath"
+
+    if (!project.hasProperty("open-report")) {
+        println "to open the report automatically in your default browser add '-Popen-report' cli argument"
+        return
+    }
+
+    def os = org.gradle.internal.os.OperatingSystem.current()
+    if (os.isWindows()) {
+        exec { commandLine 'cmd', '/c', "start $reportPath" }
+    } else if (os.isMacOsX()) {
+        exec { commandLine 'open', "$reportPath" }
+    } else if (os.isLinux()) {
+        if (localProperties.containsKey("linux-html-cmd")) {
+            exec { commandLine properties.get("linux-html-cmd"), "$reportPath" }
+        } else {
+            println "'linux-html-cmd' property could not be found in 'local.properties'"
+        }
```

`xdg-open` is what you want for most linux, I tested this (including with a bogus command to make sure it still fell back to linux-html-cmd) and it worked great

```suggestion
        try {
            exec { commandLine 'xdg-open', "$reportPath" }
        } catch (Exception ignored) {
            if (localProperties.containsKey("linux-html-cmd")) {
                exec { commandLine properties.get("linux-html-cmd"), "$reportPath" }
            } else {
                println "'linux-html-cmd' property could not be found in 'local.properties'"
            }
        }
```

---

That would be a good middle ground.

---

I did want to put `xdg-open` but I didn't know if all devs will have that installed. A fallback to the `linux-html-cmd` is a good soultion.

---

Yeah, I think most do but...linux :laughing: so you can't trust anything. Thus the fallback and I was sure to test it because...linux, you can't trust anything...